mars/dataframe/core.py

```diff
@@ -1021,6 +1021,13 @@ def index(self):
     def axes(self):
         return [self.index]
 
+    @property
+    def empty(self):
+        shape = getattr(self, '_shape')
+        if np.nan in shape:
+            raise RuntimeError("The shape is unknown until you trigger execution")
```

Comment:
`ValueError` can be raised which can refer to https://github.com/mars-project/mars/blob/6535fe1b198e2ef7b209058d442a3b3f375cfba5/mars/core/entity/executable.py#L90-L91 .